### DISCUSSION TITLE
Using the API to fix a problem with another plugin

### DISCUSSION THREAD

**Discussion by WoefieProCoder** [GriefPrevention/GriefPrevention#1678] [Q&A] [answered]
Hello  I made a plugin that stops people from placing signs if they are muted, weirdly they can place signs in a claim when they aren't muted. I want to fix that with using the api, but I dont know how. Is there an easy way to check if someone has trust in a claim? Like claim.PlayerHasTrust?

**Comment by Jikoo** [comment, marked as answer]
The only reason they'd be able to place signs in claims is if you're un-cancelling the placement event. Rather than hook into GP specifically you should probably fix your code so it isn't incompatible with every protection plugin out there.

**Reply by Jikoo** [reply]
That is to say, you should be ignoring cancelled events. My guess is you're not ignoring cancelled and you're doing something like
```Java
event.setCancelled(isMute(player));
```
This should actually be
```Java
if (isMute(player)) {
  event.setCancelled(true);
}
```

**Reply by WoefieProCoder** [reply]
this is what i have
`@EventHandler
	public void onBlockPlace( BlockPlaceEvent event) {
		
		Player player = event.getPlayer(); 
		if (event.getBlock().getState() instanceof Sign) {	
			if (alreadySignMuted(player) == true) {
				player.sendMessage(ChatColor.RED + "" + ChatColor.BOLD + "You are signmuted");
				event.setCancelled(true);
			} else {
				event.setCancelled(false);
			}
		}
	}`

**Reply by Jikoo** [reply]
Remove the else statement uncancelling the event. You are only providing additional cancellation conditions - if you uncancel the event, you will allow unmuted players to place signs all the time no matter what. You can also make your EventHandler ignore cancelled events `@EventHandler(ignoreCancelled = true)` to ignore events that have already been handled by another plugin.
